contracts/upkeeps/Cron.sol

```diff
@@ -0,0 +1,255 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.6;
+
+import {CronUtility, Spec} from "../libraries/CronUtility.sol";
+import "./UpkeepBase.sol";
+
+/**
+ * @title The Cron contract
+ * @notice A keeper-compatible contract that runs various tasks on cron schedules.
+ * Users must use the encodeCronString() function to encode their cron jobs before
+ * setting them. This keeps all the string manipulation off chain and reduces gas costs.
+ */
+contract Cron is UpkeepBase {
+  //
+  using CronUtility for Spec;
+  using CronUtility for string;
+
+  event CronJobExecuted(uint256 indexed id, uint256 timestamp);
+  event CronJobCreated(uint256 indexed id, address target, bytes handler);
+  event CronJobDeleted(uint256 indexed id);
+
+  error argumentLengthMismatch();
+  error callFailed(uint256 id, string reason);
+  error cronJobIDNotFound(uint256 id);
```

Comment:
I think our error style is inconsistent, so far it normally starts uppercase. Let's do that with the errors here too.